public/apps/account/utils.tsx

```diff
@@ -40,6 +40,12 @@ export async function logout(http: HttpStart, logoutUrl?: string): Promise<void>
     logoutUrl || `${http.basePath.serverBasePath}/app/login?nextUrl=${nextUrl}`;
 }
 
+export async function samlLogout(http: HttpStart): Promise<void> {
+  // This will ensure tenancy is picked up from local storage in the next login.
+  setShouldShowTenantPopup(null);
```

Comment:
should this be set to True instead?

Reply:
If it is set true, the tenant modal will be shown on load. [CodeRef](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/account/account-nav-button.tsx#L69)

Basic Auth flow is also handling the same by setting to null during [logout](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/account/utils.tsx#L33)